asserts/asserts.go

```diff
@@ -57,11 +57,39 @@ type AssertionType struct {
 	// PrimaryKey holds the names of the headers that constitute the
 	// unique primary key for this assertion type.
 	PrimaryKey []string
+	// OptionalPrimaryKeyDefaults holds the default values for
+	// optional primary key headers.
+	// Optional primary key headers can be added to types defined
+	// in previous versions of snapd, as long as they are added at
+	// the end of the old primary key together with a default value set in
+	// this map.
```

Comment:
Perhaps it's implicit, but maybe it should mention that there needs to be a default value for each primary key that follows the first key with a default value?